Src/ILGPU.Algorithms.Tests/RadixSortExtensionTests.tt

```diff
@@ -0,0 +1,340 @@
+        {
+            // Type, RadixSortOperation, Sequencer, Start of Sequencer,
+            // StepSize of Sequencer, BufferLength
+<#
+        var types = SignedIntTypes.Skip(2).Concat(UnsignedIntTypes.Skip(2));
+        foreach (var type in types) {
+            foreach (var size in ArraySizes) {
+#>
+            { default(<#= type.Type #>), default(XunitAscending<#= type.Name #>),
+                default(<#= type.Name #>TestSequencer), 0, 1, <#= size #> },
+<#
+            }
+        }
+#>
+        };
+
+        public static TheoryData< object, object, object, object, object, object>
+            DescendingTestData =>
+            new TheoryData< object, object, object, object, object, object>
+        {
+            // Type, RadixSortOperation, Sequencer, Start of Sequencer,
+            // StepSize of Sequencer, BufferLength
+<#
+        foreach (var type in types) {
+            foreach (var size in ArraySizes) {
+#>
+            { default(<#= type.Type #>), default(XunitDescending<#= type.Name #>),
+                default(<#= type.Name #>TestSequencer), 0, 1, <#= size #> },
+<#
+            }
+        }
+#>
+        };
+
+        public static TheoryData< object, object, object, object, object, object>
+            ConstantTestData =>
+            new TheoryData< object, object, object, object, object, object>
+        {
+            // Type, RadixSortOperation, Sequencer, Start of Sequencer,
+            // StepSize of Sequencer, BufferLength
+<#
+        foreach (var type in types) {
+            foreach (var size in ArraySizes) {
+#>
+            { default(<#= type.Type #>), default(XunitAscending<#= type.Name #>),
+                default(<#= type.Name #>TestSequencer), 0, 0, <#= size #> },
+            { default(<#= type.Type #>), default(XunitAscending<#= type.Name #>),
+                default(<#= type.Name #>TestSequencer), 1, 0, <#= size #> },
+<#
+            if (type.IsUnsignedInt) { continue; }
+#>
+            { default(<#= type.Type #>), default(XunitAscending<#= type.Name #>),
+                default(<#= type.Name #>TestSequencer), -1, 0, <#= size #> },
+<#
+            }
+        }
+#>
+        };
+
+        #endregion
+
+<#
+        string[] orders = { "Ascending", "Descending" };
+        foreach (var order in orders) {
+#>
+        [Theory]
+        [MemberData(nameof(<#= order #>TestData))]
+        public void RadixWorstCase<#= order #><T, TRadixSortOp, TSequencer>(
+            T _,
+            TRadixSortOp radixSortOp,
+            TSequencer sequencer,
+            T start,
+            T stepSize,
+            int length)
+            where T : unmanaged
+            where TRadixSortOp : struct, IRadixSortOperation<T>
+            where TSequencer : struct, ITestSequencer<T>
+        {
+            using var input = Accelerator.Allocate<T>(length);
+            using var stream = Accelerator.CreateStream();
+
+<#
+            if (order == "Ascending") {
```

Comment:
comment: this is totally fine 🤓 maybe we should change this to use an enumeration or global constants in the future